core/src/Cuda/Kokkos_Cuda_Vectorization.hpp

```diff
@@ -81,18 +81,19 @@ struct in_place_shfl_op {
     union conv_type {
       Scalar orig;
       shfl_type conv;
+      // This should be fine, members get explicitly reset, which changes the
+      // active member
+      KOKKOS_FUNCTION conv_type() { conv = 0; }
```

Comment:
For the atomics we mostly use just 
```suggestion
      KOKKOS_FUNCTION conv_type() {}
```
Just commenting for consistency.

Reply:
Yeah but the atomics use unions in a UB way I think. I tried to fix it here. We need to fix it in the atomics too.